src/test/java/hudson/plugins/ec2/util/AmazonEC2FactoryMockImpl.java

```diff
@@ -0,0 +1,68 @@
+package hudson.plugins.ec2.util;
+
+import java.net.URL;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+import hudson.Extension;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.model.DescribeRegionsResult;
+import com.amazonaws.services.ec2.model.Region;
+
+import hudson.plugins.ec2.EC2Cloud;
+
+@Extension
```

Comment:
```suggestion
@TestExtension
```

Reply:
Also, by using this annotation, you might be able to make or use a more generic `ExtensionFilter` for test extensions or something like that.